### ISSUE TITLE
warn: allow editing of template text before posting them

### ISSUE THREAD

**Issue by siddharthvp** [azatoth/twinkle#1048]
An ancient problem that's within our power to solve. This would allow users to shorten messages by cutting out what's irrelevant to the target user, and by adjusting wording as necessary. Everyone seems to agree that customised messages are better than template boilerplate. But of course they're difficult to deliver. With this change in twinkle, we'd have the best of both worlds. 

Regarding implementation, I would imagine an "edit text" link which when clicked crates a new textarea with the template text, that replaces "Optional message" textarea (with any optional message the user may have already written appended to the template text).

**Comment by Amorymeltzer** [comment]
I actually disagree that editing is a good idea.  Personalized notes are better and more effective, especially for [regulars](http://en.wikipedia.org/wiki/WP:DTTR), but for something like RCP/anti-vandal work, using standard messages ensures that the proper process goes through.  The single-notice/single-warn categories lean non-vandals, which I imagine is where any real "editing" desire comes from.  I'm open to being proven wrong, but I think most cases of wanting to issue an edited/tweaked template should probably just write what they mean.

That being said, I *could* see having an option just to issue an open-ended note from a textarea.  There are other options for sending one off messages, but it would make sense to include it under the umbrella of warnings.

**Comment by siddharthvp** [comment]
> The single-notice/single-warn categories lean non-vandals, which I imagine is where any real "editing" desire comes from.

Yes this is what I had in mind. I think it would be good to let the user decide what's convenient for them – for me personally editing sounds easier since the template boilerplate usually does contain helpful info like links to lesser-known project pages that I don't know about so I wouldn't have included if I was writing a message from scratch.
